Adapter Manuals - Offline option (#480)
ALTIBASE/Documents: Manuals/Altibase_7.3/eng/Adapter for Oracle User's Manual.md

- META_LOGGING

  This logs replication meta information and SN in the file. When a failure occurs, the file is used to configure the meta information necessary to read unapplied logs. The file path is created within the ala_meta_files folder in the log file path.

- SET OFFLINE ENABLE WITH 'log_dir' 

  This enables to use of the offline replication option. This statement can only be executed when replication is stopped. It sets up the Standby server to access the log files directly by specifying the log file path of the failed Altibase server.

- SET OFFLINE DISABLE

  This disables to use of the offline replication option. This statement can only be executed when replication is stopped.

- BUILD OFFLINE META

>   This reads meta and SN files from the ala_meta_files folder in the specified log file path. This constructs the necessary meta information for offline replication.

Author: ```suggestion
  This reads sender meta and RestartSN files from the ala_meta_files folder in the specified log file path. This constructs the necessary meta information for offline replication.
```

- RESET OFFLINE META 

  This resets the meta information configured by BUILD OFFLINE META when it is no longer needed or configuring new meta information.

- START WITH OFFLINE

  This starts replication through the specified offline path. Offline replication is a one-time operation, so it terminates right after applying all unsent logs. After the completion of offline replication, users can start replication again.

#### Constraints

- ALA can only use the read and write functions of the replication meta information file.
- The ALA object name for the server running the offline oraAdapter must be the same as the ALA object name for the Active server.
- Offline oraAdapter does not support ALA objects with compressed tables as replication targets.

> - If the offline oraAdapter cannot access the log file and meta information file paths of the Active server due to disk issues, the operation fails.

Author: ```suggestion
- If the offline oraAdapter cannot access the log file and the sender meta file paths of the Active server due to disk issues, the operation fails.
```

- The log file size of the Active server and Standby server must be the same. The size is determined during the database creation, so it must be verified before using the offline option.

> - Changing log files and meta files arbitrarily (renaming, copying log files to another system, deleting) can lead to abnormal termination issues.

Author: ```suggestion
- Changing log files and sender meta files arbitrarily (renaming, copying log files to another system, deleting) can lead to abnormal termination issues.
```
